vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -352,11 +351,10 @@ private void processGamePiece(GamePiece piece, Command command) {
     }
 
     // Piece should have a parent stack (Decks are extensions of Stacks)
-    // Except pieces with a DNS (Immobilized) trait.
+    // Except pieces that return TRUE for the NO_STACK property.
     final Stack oldStack = piece.getParent();
     if (oldStack == null) {
-      final Immobilized i = (Immobilized)Decorator.getDecorator(piece, Immobilized.class);
-      if (i == null || !i.isCanStack()) {
+      if (!Boolean.TRUE.equals(Decorator.getOutermost(piece).getProperty(Properties.NO_STACK))) {
```

Comment:
It looks like you might have a "double negative" here that you don't want? It seems to check for the *lack* of NO_STACK but then increase the noStackCount? I will admit I have lost the thread about what this little block was trying to do, but probably check back what it originally did and make sure we have the "plus" and "minus" in the right direction :)

Reply:
I believe this is correct. This is not supposed to count the non-stacking units, it is counting the number of stackable units that are not actually in a stack. So if a unit has no parent, but NO_STACK is true, then we don't want to count it.

The original code was 
    final Stack oldStack = piece.getParent();
    if (oldStack == null) {
      final Immobilized i = (Immobilized)Decorator.getDecorator(piece, Immobilized.class);
      if (i == null) {
        noStackCount++;
        log(Resources.getString("GameRefresher.refresh_error_nostack", piece.getName(), piece.getId()));
      }
    }

Reply:
Okay yes, makes sense!